### ISSUE TITLE
[BUG] "Pixel maximum menu width" not working as expected

### ISSUE THREAD

**Issue by chip33** [Hofknecht/SystemTrayMenu#319]
**Description, Steps to reproduce, Expected behavior, Screenshots**

I was hoping to put in a feature request to enable a "Pixel minimum menu width"
where you could set a minimum width and the window width would shrink to that size but expand if longer filenames were found in the directory.
This would be similar to how the Quick Launch menu works now expanding in width (to a set maximum) as the filename grows longer.
![Quick Launch](https://user-images.githubusercontent.com/48499740/152806966-fdb80f2b-4464-4d92-8bb2-0b0751194a63.png)

"Pixel maximum menu height" works as expected but changing "Pixel maximum menu width" between its limits of 50 to 1000 pixels has no visible effect on the STM window width which always looks like this.
![Screenshot 2022-02-07 143840](https://user-images.githubusercontent.com/48499740/152809312-57cc70ee-199b-43f2-83c5-dbc8a93e80a7.png)

Is this a bug in "Pixel maximum menu width" ?
If you feel that adding a  "Pixel miniimum menu width" is a good feature I could convert this to a request later.
Thank you for all your work on this project!

**Comment by Hofknecht** [comment]
With "Pixel miniimum menu width"  i am not sure, for me it is enough that minimum is smalles text and there is a fixed minimum, you can see e.g. here 
![2022-02-07 17_59_57-Window](https://user-images.githubusercontent.com/52528841/152835735-1e81278d-edf8-4908-bf9a-b404f42e1d8f.png)

related to bug in  "Pixel maximum menu width", i set e.g. 900 made a short test with 'XXXXX...' and seems to work
![2022-02-07 17_59_03-Window](https://user-images.githubusercontent.com/52528841/152835817-f0231f1e-7593-4e92-925a-4df14d826e33.png)

when i reduce to e.g. 150 it looks like this
![2022-02-07 18_05_39-Window](https://user-images.githubusercontent.com/52528841/152836403-44332439-8f92-4361-97b9-a713a019a3c3.png)

so basically should work.

What is it configured in your last screenshot and does it not change  anything when you change the value?

**Comment by chip33** [comment]
Ah yes, thank you for your screenshots they are really helpful.

At 1080p everything works perfectly and the window shrinks very narrow to fit short filenames like in your images and expands hugely to fit long filenames.

At 4k (at 300% scaling which is recommended) for some reason Windows stops the window from shrinking below about 800px 
As the maximum pixel width is 1000 there is almost no difference.
At 350% scaling the STM window never shrinks below 1000px so there is no difference at all.

I guess making the maximum width go above 1000 which is only 1/4 of the screen would show that it is working.
The not going below 800px even when set to 100px width may be a windows scaling thing rather than a setting in STM, I am not sure.

4k @ 300% display scaling 100px maximum width:

![Screenshot 2022-02-07 190314](https://user-images.githubusercontent.com/48499740/152857348-a34f4c30-47fe-4cfb-92c6-1bd02053450d.png)

4k @ 300% display scaling 1000px maximum width:

![Screenshot 2022-02-07 184404](https://user-images.githubusercontent.com/48499740/152857428-e5be713a-9e6d-4966-8024-f0620939d500.png)

4k @ 350% display scaling 100px maximum width:

![Screenshot 2022-02-07 195604](https://user-images.githubusercontent.com/48499740/152861985-f173b807-094f-44df-9228-f74c29c99be9.png)

4k @ 350% display scaling 1000px maximum width:

![Screenshot 2022-02-07 185127](https://user-images.githubusercontent.com/48499740/152862301-27277b51-f56f-4d19-a132-5c93bbd13873.png)

**Comment by Hofknecht** [comment]
@chip33 
solved with 1.2.6.4

do you want to test it and check if it works now as expected?
do you still want to configure a minimum width?

**Comment by chip33** [comment]
Thank you for looking at this.
The issue of it not going above 1000px width which is needed for 4K is fixed and tested.
The issue of it not going below ≈ 800px at 4k @ 300% display scaling and 1000px  at 4k @ 350% is still there.
I do not think a minimum width will fix this, I thought the minimum width was set too high before I found out it shrinks to fit the filename length at 1080 but not at 4K.
I do not know why it is working at 1080 but not at 4K.

**Comment by chip33** [comment]
Maybe windows is multiplying the minimum width by 350% and that is why?
In that case changing the minimum width could help.

**Comment by Hofknecht** [comment]
@chip33 
Thank you for testing.

We made 2 mistakes at the redesign 1.2
now should be solved 
can you try again?

**Comment by Hofknecht** [comment]
@chip33
If you notice any other 4k scaling problems, please feel free to create issues for them.
Even if you still want to set a minimum size you can just create an issue

**Comment by chip33** [comment]
Thank you so much, great job finding this :)
Yes, perhaps a minimum width setting would be good so users could make the window wider than the minimum if they prefer. 

Here it is working now at 4k 350%
![Screenshot 2022-02-11 201926](https://user-images.githubusercontent.com/48499740/153665190-0bf7f0d4-0891-4a6d-9f31-cee36948032a.png)
